plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/PropertyAwarePart.java

```diff
@@ -205,6 +207,21 @@ private void handleInputChange(PropertyChangeEvent evt) {
                 }
             }
 
+            List<PropertyAwarePart> childrenParts = getChildren();
+            for (Iterator<PropertyAwarePart> iter = childrenParts.iterator(); iter.hasNext();) {
+                PropertyAwarePart next = iter.next();
+                List<?> childrenConnections = next.getTargetConnections();
+                ConnectionEditPart childConnectionPartToRemove;
+                for (Iterator<?> childIter = childrenConnections.iterator(); childIter.hasNext(); ) {
+                    ConnectionEditPart childConnectionPart = (ConnectionEditPart) childIter.next();
+                    if (childConnectionPart.getModel() == oldValue) {
+                        childConnectionPartToRemove = childConnectionPart;
+                        next.removeTargetConnection(childConnectionPartToRemove);
+                        break;
+                    }
+                }
+            }
```

Comment:
What's the point of using `Iterator` here without utilizing its properties, such as `Iterator#remove`?